File: consts.go
package main

import (
	"math"
	"math/rand"
)

const (
	Radius                          = 15000                   // Game map radius
	MaxAsteroidSize                 = 50                      // Maximum size of generated asteroids
	MinAsteroidSize                 = MaxAsteroidSize / 7 * 5 // Minimum size of generated asteroids
	AsteroidCount                   = 500                     // Number of generated asteroids in the game
	WormholeCount                   = 25                      // Number of generated wormhole pairs in the game
	ShipMaxHealth                   = 100                     // Maximum health points for ships
	ShipStartFuel                   = 100                     // Starting fuel for new ships
	PlayerStartFuel                 = 1000                    // Starting fuel for players
	PlayerStartRock                 = 1000                    // Starting rock resources for players
	BaseShipMovementFree            = 1.0                     // Free movement distance before fuel cost
	BaseShipMovementMultiplier      = 1.0                     // Fuel cost multiplier for movement delta beyond free range
	ShipMovementMaxSize             = 10000                   // Maximum movement delta per turn - larger movements are scaled down
	ShipTransferDistance            = 20                      // Maximum distance for resource transfer between ships
	ShipShootDistance               = 500                     // Maximum shooting range for ships
	ShipShootDamage                 = 25                      // Damage dealt by ship weapons
	ShipRepairDistance              = 50                      // Maximum distance for ship repair operations
	ShipRepairAmount                = 30                      // Health points restored by repair
	ShipRepairRockCost              = 15                      // Rock cost per repair operation
	MaterialToSurfaceRatio          = 0.1                     // Ratio of material to surface area for asteroids
	AsteroidSpawnOffset             = 40.0                    // Offset distance for asteroid spawning after ship death
	GlobalAsteroidMovementScale     = 2.0                     // Global scale factor for asteroid movement
	IndividualAsteroidMovementScale = 1.0                     // Individual asteroid movement scale factor
	PerlinNoiseScale                = 0.01                    // Scale factor for Perlin noise generation
	WormholeRadius                  = 5                       // Radius within which ships get teleported by wormholes
	WormholeTeleportDistance        = WormholeRadius * 2      // Minimum distance from target wormhole (2x radius) to prevent teleport loops
	ShipMiningDistance              = MaxAsteroidSize         // Maximum distance for mining operations
	ShipMiningAmount                = 10                      // Units mined per tick
	ShipConqueringDistance          = MaxAsteroidSize         // Maximum distance for conquering operations
	ShipConqueringRate              = 10                      // Surface units conquered/lost per tick
)

func ShipRockPrice(t ShipType) int {
	return 250
}

func ShipMovementFree(t ShipType) float64 {
	switch t {
	case TruckShip:
		fallthrough
	case TankerShip:
		return BaseShipMovementFree * 3
	default:
		return BaseShipMovementFree
	}
}

func ShipMovementMultiplier(t ShipType) float64 {
	switch t {
	case TruckShip:
		fallthrough
	case TankerShip:
		return BaseShipMovementMultiplier / 3.0
	default:
		return BaseShipMovementMultiplier
	}
}

func ShipMovementPrice(vector Position, t ShipType) float64 {
	return max(0.0, (vector.Size()-ShipMovementFree(t))*ShipMovementMultiplier(t))
}

func RandomFloat(min, max float64) float64 {
	return rand.Float64()*(max-min) + min
}

func AsteroidScore(a Asteroid) float64 {
	asteroidSurface := float64(a.Size * a.Size * math.Pi)
	ownedSurface := float64(a.OwnedSurface)
	surfaceFactor := math.Pow(1.5, (ownedSurface/asteroidSurface*100)/9.0) * (a.Size / float64(MaxAsteroidSize))
	return 50 + surfaceFactor
}

File: rust_bot/src/proboj/consts.rs
pub const RADIUS: f64 = 15000.0; // Game map radius

pub const MAX_ASTEROID_SIZE: f64 = 50.0; // Maximum size of generated asteroids
pub const MIN_ASTEROID_SIZE: f64 = MAX_ASTEROID_SIZE / 7.0 * 5.0; // Minimum size of generated asteroids

pub const ASTEROID_COUNT: f64 = 500.0; // Number of generated asteroids in the game
pub const WORMHOLE_COUNT: f64 = 25.0; // Number of generated wormhole pairs in the game

pub const WORMHOLE_RADIUS: f64 = 5.0; // Radius within which ships get teleported by wormholes
pub const WORMHOLE_TELEPORT_DISTANCE: f64 = WORMHOLE_RADIUS * 2.0; // Minimum distance from target wormhole (2x radius) to prevent teleport loops

pub const SHIP_MINING_DISTANCE: f64 = MAX_ASTEROID_SIZE; // Maximum distance for mining operations
pub const SHIP_MINING_AMOUNT: f64 = 10.0; // Units mined per tick
pub const SHIP_CONQUERING_DISTANCE: f64 = MAX_ASTEROID_SIZE; // Maximum distance for conquering operations
pub const SHIP_CONQUERING_RATE: f64 = 10.0; // Rate of conquering

pub const SHIP_MAX_HEALTH: f64 = 100.0; // Maximum health points for ships
pub const SHIP_START_FUEL: f64 = 100.0; // Starting fuel for new ships
pub const SHIP_MOVEMENT_FREE_SIZE: f64 = 1.0; // Movement delta per turn that is free of fuel cost
pub const SHIP_MOVEMENT_MAX_SIZE: f64 = 10000.0; // Maximum movement delta per turn - larger movements are scaled down
pub const SHIP_TRANSFER_DISTANCE: f64 = 20.0; // Maximum distance for resource transfer between ships
pub const SHIP_SHOOT_DISTANCE: f64 = 500.0; // Maximum shooting range for ships
pub const SHIP_SHOOT_DAMAGE: f64 = 25.0; // Damage dealt by ship weapons
pub const SHIP_REPAIR_DISTANCE: f64 = 50.0; // Maximum distance for ship repair operations
pub const SHIP_REPAIR_AMOUNT: f64 = 30.0; // Health points restored by repair
pub const SHIP_REPAIR_ROCK_COST: f64 = 15.0; // Rock cost per repair operation
